sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -187,12 +190,14 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
       RefreshTableExec(r.catalog, r.identifier, recacheTable(r)) :: Nil
 
     case ReplaceTable(ResolvedDBObjectName(catalog, ident), schema, parts, tableSpec, orCreate) =>
+      val newSchema: StructType =
+        ResolveDefaultColumns.constantFoldCurrentDefaultsToExistDefaults(schema, "CREATE TABLE")
```

Comment:
I suggest passing the analyzer via function parameter
```
session.sessionState.analyzer
```
instead of creating a new one in object `ResolveDefaultColumns`
After supporting multiple catalogs, different catalogs will have different SQL functions which can be used in default values.

Reply:
Nice, this is more consistent. Thank you for suggesting this.